### PR TITLE
Changes for Nginx

### PR DESCRIPTION
**Pull request by SparkiDev**
Support TLS v1.3 clients connecting to Nginx.
Fix for PSS to not advertise hash unless the signature fits the private
key size.
Allow curves to be chosen by user.
Support maximum verification depth (maximum number of untrusted certs in
chain.)
Add support for SSL_is_server() API.
Fix number of certificates in chain when using
wolfSSL_CTX_add_extra_chain_cert().
Allow TLS v1.2 client hello parsing to call TLS v1.3 parsing when
SupportedVersions extension seen.
Minor fixes.

### CODE REVIEW

File: src/tls13.c [hunk 1]
```diff
@@ -3295,10 +3296,14 @@ static int DoTls13ClientHello(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
 #endif
 
     /* Session id - empty in TLS v1.3 */
-    b = input[i++];
-    if (b != 0) {
-        WOLFSSL_MSG("Client sent session id - not supported");
-        return BUFFER_ERROR;
+    sessIdSz = input[i++];
+    if (sessIdSz > 0) {
```
Comment (toddouska) [inline src/tls13.c]: Why isn't this an error if it's not allowed?

File: src/ssl.c [hunk 2]
```diff
@@ -380,6 +380,11 @@ void wolfSSL_free(WOLFSSL* ssl)
 }
 
 
+int wolfSSL_is_server(WOLFSSL* ssl)
+{
+    return ssl->options.side == WOLFSSL_SERVER_END;
```
Comment (toddouska) [inline src/ssl.c]: Let's add a NULL check first.

File: src/ssl.c [hunk 3]
```diff
@@ -24958,6 +25008,32 @@ void wolfSSL_get0_next_proto_negotiated(const WOLFSSL *s, const unsigned char **
 }
 #endif /* HAVE_ALPN */
 
+WOLFSSL_API int wolfSSL_CTX_set1_curves_list(WOLFSSL_CTX* ctx, char* names)
```
Comment (toddouska) [inline src/ssl.c]: Why are only 3 curves detected?  Can you add a little documentation to this function please?

Comment (SparkiDev) [inline src/ssl.c]: Changed to use wc_ecc_get_curve_id_from_name() and added some different versions recognized by OpenSSL.

File: src/ssl.c (hunk 2 above)
Comment (SparkiDev) [inline src/ssl.c]: Fixed

File: src/tls13.c (hunk 1 above)
Comment (SparkiDev) [inline src/tls13.c]: Undid this as it wasn't needed after all.

**Pull request merged**